src/app/core/auth.service.spec.ts

```diff
@@ -1,6 +1,92 @@
+import { async, TestBed } from '@angular/core/testing';
+import { Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { OAuthService } from 'angular-oauth2-oidc';
+import { MockOAuthService } from '../../testing/mocks';
+import { AuthService } from './auth.service';
 
 describe('AuthService', () => {
-  it('should pass', () => {
-    expect(true).toBeTruthy();
+  let service: AuthService;
+  let mockService: MockOAuthService;
+  let router: Router;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [RouterTestingModule],
+      providers: [
+        AuthService,
+        {provide: OAuthService, useClass: MockOAuthService}
+      ]
+    });
+
+    service = TestBed.inject(AuthService);
+    mockService = <any>TestBed.inject(OAuthService) as MockOAuthService;
+    router = TestBed.inject(Router);
+    spyOn(router, 'navigateByUrl').and.stub();
+  });
+
+  it('should react on OAuthService events', () => {
+    spyOn(mockService, 'hasValidAccessToken');
+    spyOn(mockService, 'loadUserProfile');
+
+    mockService.emulateEvent({type: 'silently_refreshed'});
+    mockService.emulateEvent({type: 'token_received'});
+
+    expect(mockService.loadUserProfile).toHaveBeenCalled();
+    expect(mockService.hasValidAccessToken).toHaveBeenCalledTimes(2);
+
+    mockService.emulateEvent({type: 'session_terminated'});
+    mockService.emulateEvent({type: 'session_error'});
```

Comment:
It might be a good improvement to have the enumeration of options outside the spec, so:

```typescript
  ['session_terminated', 'session_error'].forEach(eventType => {
    it(`should react on OAuthService event ${eventType}`, () => {
      // ...
      mockService.emulateEvent({type: <EventType>eventType});
      // ...
    });
  });
```